### ISSUE TITLE
Missing keywords for new java features

### ISSUE THREAD

**Issue by gayanper** [eclipse-jdt/eclipse.jdt.core#2255]
The following keywords are missing in Code Completions

- permits
- record

**Comment by gayanper** [comment]
@stephan-herrmann @mpalat was there a reason we didn't add them into code completion keywords ? If not I can make a PR adding them.

**Comment by stephan-herrmann** [comment]
permits and record look simple, as those don't need a lot of context :)

Sure a PR would be welcome from my p.o.v.

**Comment by mpalat** [comment]
> @stephan-herrmann @mpalat was there a reason we didn't add them into code completion keywords ? If not I can make a PR adding them.

No, there was no reason other than lack of hands and priority. Wholeheartedly welcome if you can add the restricted identifiers.

**Comment by gayanper** [comment]
I think part of this issue is fixed by https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2261. I will check rest of the keywords when I get some more time to spend
